manifests/00-crd-extension-console-notification-banner.yaml

```diff
@@ -0,0 +1,72 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  # TODO: consolenotificationbanners.console.openshift.io
+  # will we ever have consolenotifications that act like
+  # traditional notifications?
+  name: consolenotifications.console.openshift.io
+  annotations:
+    displayName: ConsoleNotification
+    description: Extension for configuring openshift web console notifications.
+spec:
+  # TODO:
+  # this relates to where the API resources live in openshift/api
+  # likely under /config/console, which might make it something like:
+  # extensions.console.config.openshift.io?
+  group: console.openshift.io
+  version: v1
+  versions:
+  - name: v1
+    served: true
+    storage: true
+  scope: Cluster
+  names:
+    plural: consolenotifications
+    singular: consolenotification
+    kind: ConsoleNotification
+    listKind: ConsoleNotificationList
+  additionalPrinterColumns:
+  - name: Text
+    type: string
+    JSONPath: .spec.text
+  - name: Age
+    type: date
+    JSONPath: .metadata.creationTimestamp
+  subresources:
+    status: {}
+  validation:
+    openAPIV3Schema:
+      properties:
+        spec:
+          type: object
+          description: Represents console notification spec
+          required:
+          - text
+          - position
+          properties:
+            text:
+              type: string
+              description: Text of the notification
+            position:
+              type: string
+              description: The location of the notification (top, bottom, topBottom)
+              pattern: '\btop\b|\bbottom\b|\btopBottom\b'
+            link:
+              type: object
+              description: Object that holds notification link details
+              required:
+              - href
+              properties:
+                text:
+                  type: string
+                  description: Text of the link
+                href:
+                  type: string
+                  description: Absolute URL for the link
+                  pattern: '^http(s)?://([\w-]+.)+[\w-]+(/[\w- ./?%&=])?$'
+            backgroundColor:
+              type: string
```

Comment:
Should this clarify if it needs to be a supported CSS color name or a hex value?

Reply:
Also, we should make it optional & defaulted.

Reply:
https://github.com/openshift/console/pull/1542#discussion_r285668581